slackclient/server.py

```diff
@@ -130,7 +130,17 @@ def rtm_connect(self, reconnect=False, timeout=None, use_rtm_start=True, **kwarg
         reply = self.api_requester.do(self.token, connect_method, timeout=timeout, post_data=kwargs)
 
         if reply.status_code != 200:
-            raise SlackConnectionError(reply=reply)
+            if 'depth' in kwargs:
```

Comment:
i've never heard of a `depth` argument. is this new? does it need to be documented?

Reply:
This could probably be handled better. It's to track how many attempts have been made

Reply:
Maybe just use a loop...?

Reply:
@harlowja can you elaborate on what that might look like? 🤔

Reply:
Something like:

```
reply = self.api_requester.do(self.token,
                              connect_method, timeout=timeout,
                              post_data=kwargs)
retry_attempt = 0
while reply.status_code != 200:
    if reply.status_code == 429:
        ratelimit_timeout = int(reply.headers.get('retry-after', 120))
        time.sleep(ratelimit_timeout)
    retry_attempt += 1
    if retry_attempt > 10:
        raise SlackConnectionError("RTM connection attempt was rate limited 10 times.")
    else:
        reply = self.api_requester.do(self.token,
                              connect_method, timeout=timeout,
                              post_data=kwargs)
```

You can also use http://tenacity.readthedocs.org/ and it will do alot of this for u.

Reply:
I updated it to work more like the `reconnect_count` login in the block above the retry logic

Reply:
tenacity looks like it'd be a very good addition to the work I have planned for 2.x

Reply:
Ok, I am co-author of that library, so feel free to ask questions.